src/posix/platform/openthread-posix-config.h

```diff
@@ -140,10 +140,22 @@
 #define OPENTHREAD_POSIX_CONFIG_SECURE_SETTINGS_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_POSIX_CONFIG_NETIF_LINK_LOCAL_ROUTE_METRIC
+ *
+ * This setting configures the link-local route metric on the Thread network interface.
+ * Define as 0 to use use the default prefix route metric.
+ *
+ * Note: The feature works on Linux kernel v4.18+.
+ */
+#ifndef OPENTHREAD_POSIX_CONFIG_NETIF_LINK_LOCAL_ROUTE_METRIC
+#define OPENTHREAD_POSIX_CONFIG_NETIF_LINK_LOCAL_ROUTE_METRIC 0
```

Comment:
IPv6 link-local addresses should be scoped to a specific network interface. I believe sending to a link-local address should not be routed purely on the IPv6 address itself, but also requires specifying the network interface when sending. I'm curious, is the mDNS daemon in question not implemented properly?

Reply:
>  I'm curious, is the mDNS daemon in question not implemented properly?

Yes. The mDNS daemon should bind to the network interface or using policy routing when it's sending a response.

However, Linux doesn't drop/reject such outgoing packets. Instead, they just route them like other packets. 

E.g. `ip route get fe80::1234` will return a route depending on the order of routes in the table.